flink-dist/src/main/flink-bin/bin/stop-cluster.sh

```diff
@@ -25,14 +25,30 @@ bin=`cd "$bin"; pwd`
 # Stop TaskManager instance(s) using pdsh (Parallel Distributed Shell) when available
 readSlaves
 
-command -v pdsh >/dev/null 2>&1
-if [[ $? -ne 0 ]]; then
+# all-local setup?
+all_localhost=1
+for slave in ${SLAVES[@]}; do
+    if [[ "$slave" != "localhost" ]]; then
+        all_localhost=0
+        break
+    fi
+done
```

Comment:
makes sense - also since there we already go through the slaves once so we can do this check on the fly